optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/calendar/Drawable.java

```diff
@@ -3,38 +3,128 @@
 import elemental2.dom.CanvasRenderingContext2D;
 import elemental2.dom.MouseEvent;
 
+/**
+ * Something that can be drawn to a canvas.
+ */
 public interface Drawable {
 
+    /**
+     * Draws the thing.
+     * @param g Where to draw the thing to.
+     */
     public void draw(CanvasRenderingContext2D g);
 
+    /**
+     * Returns the parent of this Drawable (can be null).
+     * @return The parent of this Drawable.
+     */
     public Drawable getParent();
 
+    /**
+     * Sets the parent of this Drawable (can be null).
+     * @param parent The new parent of this Drawable.
+     */
     public void setParent(Drawable parent);
 
+    /**
+     * The x-position of this Drawable realitive to its parent.
+     * @return The x-position of this Drawable realitive to its parent.
+     */
     public double getLocalX();
 
+    /**
+     * The y-position of this Drawable realitive to its parent.
+     * @return The y-position of this Drawable realitive to its parent.
+     */
     public double getLocalY();
 
+    /**
+     * The global x-position of this Drawable.
+     * @return The global x-position of this Drawable.
+     */
     public double getGlobalX();
 
+    /**
+     * The global y-position of this Drawable.
+     * @return The global y-position of this Drawable.
+     */
     public double getGlobalY();
 
-    //These events return true if the event is consumed
+    //These events return true if the event is consumed. Additionally,
+    //they are only called if the mouse is over the Drawable.
+
+    /**
+     * Called when the mouse enters the Drawable.
```

Comment:
To give another data-point, this (and all the other even handler method docs in this file) are all good javadocs that concisely provide useful information not communicated by the method and parameter names.